code-challenges.js

```diff
@@ -2,50 +2,121 @@
 
 // --------------------1) Create a function that takes in an array. Each time the function is run, remove the first item from the array and shuffle the remaining content. If the array is empty, return "The array is empty."
 
-var collections = ["purple", "blue", "green", "yellow", "pink"]
+var collections = ['purple', 'blue', 'green', 'yellow', 'pink']
 // Expected output example1 (can be a different order): ["yellow", "blue", "pink", "green"]
 // Expected output example2 (can be a different order): ["blue", "green", "pink"]
 // Expected output example3 (can be a different order): ["pink", "green"]
 
-
-
-
+// Create a function that takes an array as an argument
+const arrayShuffler = (arr) => {
+	// let randomNum = Math.floor(Math.random() * arr.length)
+	// console.log('random number', randomNum)
+	// check if the array is empty or not
+	// if not empty remove first item from array
+	if (arr.length > 0) {
+		arr.shift()
+		// return random item from array
+		// ****** NOTE: I FOUND THIS NEXT LINE ON FREECODECAMP BUT I DON'T THINK IT'S WORKING QUITE RIGHT ******
+		return arr.sort((a, b) => 0.5 - Math.random())
+	} else {
+		return 'The array is empty.'
+	}
+}
+
+// call the function
+console.log('Problem #1')
+console.log(arrayShuffler(collections))
+console.log(arrayShuffler(collections))
+console.log(arrayShuffler(collections))
+console.log(arrayShuffler(collections))
+console.log(arrayShuffler(collections))
+console.log(arrayShuffler(collections))
 
 // --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
-var cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-var cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
-
-
-
-
+var cubeAndSum1 = [2, 3, 4] // Expected output: 99
+var cubeAndSum2 = [0, 5, 10] // Expected output: 1125
+
+// Create a function that takes an array of numbers as a argument
+const cubeAndSum = (arr) => {
+	// create a variable to hold the accumulated value
+	let result = 0
+	// loop through array and cube each number and add to accumulating result value
+	arr.map((value) => {
+		return (result = value * value * value + result)
```

Comment:
good work! this can also be shortened to `return (result = value ** 3 + result)`